### PR TITLE
Tasks controller #41

### PR DESCRIPTION
**Pull request by pospieszynski**
tasks controller + bunch of corresponding specs.
feedback pls;]

### CODE REVIEW

File: app/helpers/tasks_helper.rb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+module TasksHelper
```
Comment (neonowy) [inline app/helpers/tasks_helper.rb]: I don't think we need an empty helper.

File: spec/controllers/tasks_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,392 @@
+require 'rails_helper'
+
+RSpec.describe TasksController, type: :controller do
+  let(:task) { FactoryGirl.create(:task) }
+  let(:another_task) { FactoryGirl.create(:task) }
+  let(:project) { task.project }
+  let(:user) { project.user }
+
+  describe "when suer is not authenticated" do
```
Comment (neonowy) [inline spec/controllers/tasks_controller_spec.rb]: Typo, `suer`.

File: spec/controllers/tasks_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+  end
+
+  context "when user is authenticated" do
+    before { sign_in(user) }
+
+    describe "#index" do
+      context "with proper params" do
+        before { get :index, params: { project_id: project.id } }
+
+        it "assigns tasks" do
```
Comment (neonowy) [inline spec/controllers/tasks_controller_spec.rb]: Imho this description is a little confusing. `#index` action doesn't assign tasks to someone. Maybe `passes @tasks to view` or something like that would be better.

File: spec/controllers/tasks_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+
+        it "return descriptive flush message" do
+          expect(flash[:notice]).to include("Error, wrong params in the request - record could not be found")
+        end
+      end
+    end
+
+    describe "#show" do
+      context "with proper params" do
+        context "when task and project are associated " do
```
Comment (neonowy) [inline spec/controllers/tasks_controller_spec.rb]: Unnecessary space after `associated`.

File: app/controllers/tasks_controller.rb [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  def set_project
+    @project = Project.find(params[:project_id])
+  end
+
+  def set_task
+    @task = Task.find(params[:id])
+  end
+
+  def check_if_params_match
+    redirect_to root_path, notice: "Params don't match" unless @project.tasks.include?(@task)
```
Comment (neonowy) [inline app/controllers/tasks_controller.rb]: Umm, maybe just render 404 page? _Params don't match_ doesn't tell user much imho.

File: spec/controllers/tasks_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,392 @@
...
+
+    describe "#create" do
+      context "with valid params" do
+        before { post :create, params: { project_id: project.id, task: FactoryGirl.attributes_for(:task) } }
+
+        it "creates task in memory initialized with a gives params" do
+          expect(assigns(:task)).to eql(Task.last)
+        end
+
+        it "redircets to task show page after it was successfully saved in database" do
```
Comment (neonowy) [inline spec/controllers/tasks_controller_spec.rb]: Typo, `redircets`.

File: spec/controllers/tasks_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+          expect(response).to redirect_to(project_task_path(Task.last.project, Task.last))
+        end
+
+        it "shows successfull flush message " do
```
Comment (neonowy) [inline spec/controllers/tasks_controller_spec.rb]: Unnecessary space after `message` and typo in `successfull`.

File: spec/controllers/tasks_controller_spec.rb [hunk 8]
```diff
@@ -0,0 +1,392 @@
...
+          expect(flash[:notice]).to include("Task was successfully created")
+        end
+      end
+
+      context "with invalid params" do #?
```
Comment (neonowy) [inline spec/controllers/tasks_controller_spec.rb]: `#?` ?

File: spec/factories/tasks.rb [hunk 9]
```diff
@@ -2,7 +2,7 @@
   factory :task do
     title Faker::Lorem.sentence(2, true, 0)
     description Faker::Lorem.sentence
-    due_date Faker::Time.forward(10)
+    due_date Faker::Time.forward(100)
```
Comment (neonowy) [inline spec/factories/tasks.rb]: Umm, why?

File: app/controllers/tasks_controller.rb (hunk 5 above)
Comment (pospieszynski) [inline app/controllers/tasks_controller.rb]: I see your point but I'm not sure if it's worth changing now, 404 page looks scary for regular user and it's even less descriptive imho. ;]

File: app/assets/javascripts/tasks.js [hunk 10]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
+// All this logic will automatically be available in application.js.
```
Comment (gotar) [inline app/assets/javascripts/tasks.js]: rely? still, drop not used files

File: app/assets/stylesheets/tasks.scss [hunk 11]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the tasks controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: http://sass-lang.com/
```
Comment (gotar) [inline app/assets/stylesheets/tasks.scss]: drop not used file

**Comment by filip373** [comment]
:+1:

File: spec/controllers/tasks_controller_spec.rb [hunk 12]
```diff
@@ -0,0 +1,392 @@
...
+        it "renders new template" do
+          expect(response).to render_template(:new)
+        end
+        it "has a 200 status code" do
+          expect(response.status).to eq(200)
+        end
+      end
+
+      context "with improper params" do
+        before { get :new, params: { project_id: 123 } }
```
Comment (tprochniewicz) [inline spec/controllers/tasks_controller_spec.rb]: I wouldn't use fixed number, I think that 'Project.last.id + 1' would be better

**Comment by tprochniewicz** [comment]
@gotar

**Comment by gotar** [comment]
good enough

**Comment by gotar** [comment]
you can merge

**Pull request merged**